src/js/content_scripts/gamepad-navigator.js

```diff
@@ -0,0 +1,119 @@
+                // Scan the state of gamepad frequently.
+                that.options.connectivityIntervalReference = setInterval(function () {
+                    // Catch the current state of gamepad.
+                    var connectedGamepad = navigator.getGamepads()[0];
+
+                    // Initiate the gamepad navigator model transaction.
+                    var modelUpdateTransaction = that.applier.initiate();
+
+                    modelUpdateTransaction.fireChangeRequest({ path: "index", value: connectedGamepad.index });
+                    modelUpdateTransaction.fireChangeRequest({ path: "axes", value: that.transformArrayToObject(connectedGamepad.axes) });
+                    modelUpdateTransaction.fireChangeRequest({ path: "buttons", value: that.transformArrayToObject(connectedGamepad.buttons) });
+
+                    // Commit the current state of gamepad.
+                    modelUpdateTransaction.commit();
+                }, that.options.frequency);
+            }
+        });
+
+        // When gamepad is disconnected
+        window.addEventListener("gamepaddisconnected", function (event) {
+            // Ensure that the gamepad navigator doesn't break if any other gamepad is disconnected.
+            if (that.model.index === event.gamepad.index) {
+                // Stop the interval loop scanning the gamepad state.
+                clearInterval(that.options.connectivityIntervalReference);
+
+                // Restore the gamepad navigator model to its initial state.
+                var modelUpdateTransaction = that.applier.initiate();
+
+                modelUpdateTransaction.fireChangeRequest({ path: "index", value: null });
+                modelUpdateTransaction.fireChangeRequest({ path: "axes", value: [] });
+                modelUpdateTransaction.fireChangeRequest({ path: "buttons", value: [] });
+
+                // Commit the initial model.
+                modelUpdateTransaction.commit();
+            }
+        });
+    };
+
+    /**
+     *
+     * Converts the input array into an object with the keys as the index of the
+     * corresponding element.
+     *
+     * @param {Array} array - The buttons and axes array to be converted into an object.
+     * @return {Object} outputObject - The transformed object.
+     *
+     */
+    gamepad.gamepadNavigator.arrayToObject = function (array) {
+        var outputObject = {};
+        fluid.each(array, function (inputData, index) {
+            /**
+             * Button data is stored as an array of objects whereas axes data is stored
+             * as an array of floating numbers. To obtain the primary values, we need to
+             * access both differently.
+             */
+            outputObject[index] = fluid.isPrimitive(inputData) ? inputData : inputData.value;
+        });
+        return outputObject;
+    };
+
+    // Create an instance of the gamepad navigator.
+    gamepad.gamepadNavigator();
```

Comment:
I'm happy for this to just be `gamepad.navigator`.

Reply:
Yes, I think that's better and also convenient for reference to this component. I'll make this change in my next commit.